support/android/apk/servoapp/src/main/java/com/mozilla/servo/MainActivity.java

```diff
@@ -173,4 +173,12 @@ public void onResume() {
         super.onResume();
     }
 
+    @Override
+    public void onBackPressed() {
+        if (mBackButton.isEnabled()) {
```

Comment:
Instead of using the button state, can you introduce a boolean field `mCanGoBack` that you set to true/false on `onHistoryChanged`?